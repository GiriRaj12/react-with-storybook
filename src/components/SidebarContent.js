import React from 'react'

function SidebarContent(props) {

    return (
        <a className="sidebar-content" href={props.toRe}>
            {props.renderIcon}
            <span className="sidebar-content-name">{props.content}</span>
        </a>)
}
export default SidebarContent;