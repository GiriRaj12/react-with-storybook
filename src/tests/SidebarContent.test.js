import React from 'react';
import { render } from '@testing-library/react';
import SidebarContent from '../components/SidebarContent.js';
import renderer from 'react-test-renderer';


test('Sidebar content test', () => {
    const component = renderer.create(
        <SidebarContent toRe="/dashboard" content="Something" renderIcon="S"></SidebarContent>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test("Side bar content name to be in the document", () => {
    const { getByText } = render(<SidebarContent toRe="/dashboard" content="Something" renderIcon="S" ></SidebarContent>);
    let element = getByText(/Something/i);
    expect(element.classList.contains('sidebar-content-name')).toBe(true);
});

test("Side bar content to be in the document", () => {
    const { container } = render(<SidebarContent toRe="/dashboard" content="Something" renderIcon="S" ></SidebarContent>);
    expect(container.firstChild.classList.contains('sidebar-content')).toBe(true);
});
