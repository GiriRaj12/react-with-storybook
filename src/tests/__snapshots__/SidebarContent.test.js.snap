// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar content test 1`] = `
<a
  className="sidebar-content"
  href="/dashboard"
>
  S
  <span
    className="sidebar-content-name"
  >
    Something
  </span>
</a>
`;
